Branch-SDK/Branch-SDK/ContentDiscoveryManifest.m

```diff
@@ -0,0 +1,107 @@
+//
+//  ContentDiscoverManifest.m
+//  Branch-TestBed
+//
+//  Created by Sojan P.R. on 8/18/16.
+//  Copyright © 2016 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "ContentDiscoveryManifest.h"
+#import "BNCPreferenceHelper.h"
+#import "ContentPathProperties.h"
+#import <UIKit/UIKit.h>
+#import "BranchConstants.h"
+
+@implementation ContentDiscoveryManifest
+
+NSString *manifestVersion;
+static ContentDiscoveryManifest *contentDiscoveryManifest;
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        NSDictionary *savedManifest = [[BNCPreferenceHelper preferenceHelper]getContentAnalyticsManifest];
+        if (savedManifest != nil ) {
+            _cdManifest = [savedManifest mutableCopy];
+        }
+        else {
+            _cdManifest = [[NSMutableDictionary alloc] init];
+        }
+    }
+    return self;
+}
+
++ (ContentDiscoveryManifest *)getInstance {
+    if (!contentDiscoveryManifest) {
+        contentDiscoveryManifest = [[ContentDiscoveryManifest alloc] init];
+    }
+    return contentDiscoveryManifest;
+}
+
+
+
+- (void)onBranchInitialised:(NSDictionary *)branchInitDict withUrl:(NSString *)referredUrl {
+    _referredLink = referredUrl;
+    if ([branchInitDict objectForKey:CONTENT_DISCOVER_KEY] != nil) {
+        _isCDEnabled = YES;
+        NSDictionary *cdManifestDict = [branchInitDict objectForKey:CONTENT_DISCOVER_KEY];
+        
+        if ([cdManifestDict objectForKey:MANIFEST_VERSION_KEY] != nil) {
+            manifestVersion = [cdManifestDict objectForKey:MANIFEST_VERSION_KEY];
+        }
+        
+        if ([cdManifestDict objectForKey:MAX_VIEW_HISTORY_LENGTH] != nil) {
+            _maxViewHistoryLength = [[cdManifestDict objectForKey:MAX_VIEW_HISTORY_LENGTH] integerValue];
+        }
+        
+        if ([cdManifestDict objectForKey:MANIFEST_KEY] != nil) {
+            _contentPaths = [cdManifestDict objectForKey:MANIFEST_KEY];
+        }
+        
+        if ([cdManifestDict objectForKey:MAX_TEXT_LEN_KEY] != nil) {
+            _maxTextLen = [[cdManifestDict objectForKey:MAX_TEXT_LEN_KEY]integerValue];
+        }
+        
+        if ([cdManifestDict objectForKey:MAX_PACKET_SIZE_KEY] != nil) {
+            _maxPktSize = [[cdManifestDict objectForKey:MAX_PACKET_SIZE_KEY]integerValue];
+        }
+        
+        [_cdManifest setObject:manifestVersion forKey:MANIFEST_VERSION_KEY];
+        [_cdManifest setObject:_contentPaths forKey:MANIFEST_KEY];
```

Comment:
seems like it's possible for both of these to be null if something on the backend is messed up. Should you just set these in their respective if statement that handles null checking?